File: example_usage.js
const axios = require("axios");

// Example usage of the Stock Availability API

// API endpoint
const API_URL = "http://localhost:8000/api/stock-availability";

// Example 1: Basic search
const example1 = {
  city: "chennai",
  skus: ["SKU4"],
  page: 1,
  pageSize: 2,
  sortBy: "TOTAL STOCK",
  sortOrder: "DESC",
  dateFrom: "2025-09-01",
  dateTo: "2025-09-30",
  search: "toothbrush",
};

// Example 2: Search without city filter
const example2 = {
  skus: ["SKU4", "SKU5"],
  page: 1,
  pageSize: 10,
  sortBy: "INSTOCK DARKSTORES",
  sortOrder: "ASC",
  search: "baby",
};

// Example 3: Search with date range only
const example3 = {
  page: 1,
  pageSize: 5,
  sortBy: "DAYS OF STOCK",
  sortOrder: "DESC",
  dateFrom: "2025-09-10",
  dateTo: "2025-09-20",
  search: "cleaning",
};

// Example 4: Delhi baby wipes search
const example4 = {
  city: "delhi",
  skus: ["SKU1", "SKU2"],
  page: 1,
  pageSize: 2,
  sortBy: "TOTAL STOCK",
  sortOrder: "DESC",
  dateFrom: "2025-09-01",
  dateTo: "2025-09-30",
  search: "baby wipes",
};

async function testAPI(exampleData, exampleName) {
  console.log(`\n=== ${exampleName} ===`);
  console.log("Request JSON:");
  console.log(JSON.stringify(exampleData, null, 2));

  try {
    const response = await axios.post(API_URL, exampleData, {
      headers: {
        "Content-Type": "application/json",
      },
      timeout: 10000,
    });

    console.log(`\nResponse Status: ${response.status}`);
    console.log("Response JSON:");
    console.log(JSON.stringify(response.data, null, 2));
  } catch (error) {
    if (error.code === "ECONNREFUSED") {
      console.log(
        "Error: Could not connect to the API. Make sure the server is running on http://localhost:8000"
      );
    } else if (error.response) {
      console.log(
        `Error: ${error.response.status} - ${
          error.response.data.error || error.response.data.message
        }`
      );
    } else {
      console.log(`Error: ${error.message}`);
    }
  }
}

async function runTests() {
  console.log("Stock Availability API Test Examples");
  console.log("=".repeat(50));

  // Test all examples
  await testAPI(example1, "Example 1: Chennai toothbrush search");
  await testAPI(example2, "Example 2: Baby products search");
  await testAPI(example3, "Example 3: Cleaning products with date range");
  await testAPI(example4, "Example 4: Delhi baby wipes search");

  console.log("\n" + "=".repeat(50));
}

// Run if called directly
if (require.main === module) {
  runTests().catch(console.error);
}

module.exports = { testAPI, runTests };

File: example_usage.py
import requests
import json

# Example usage of the Stock Availability API

# API endpoint
url = "http://localhost:8000/api/stock-availability"

# Example 1: Basic search
example_1 = {
    "city": "chennai",
    "skus": ["SKU4"],
    "page": 1,
    "pageSize": 2,
    "sortBy": "TOTAL STOCK",
    "sortOrder": "DESC",
    "dateFrom": "2025-09-01",
    "dateTo": "2025-09-30",
    "search": "toothbrush"
}

# Example 2: Search without city filter
example_2 = {
    "skus": ["SKU4", "SKU5"],
    "page": 1,
    "pageSize": 10,
    "sortBy": "INSTOCK DARKSTORES",
    "sortOrder": "ASC",
    "search": "baby"
}

# Example 3: Search with date range only
example_3 = {
    "page": 1,
    "pageSize": 5,
    "sortBy": "DAYS OF STOCK",
    "sortOrder": "DESC",
    "dateFrom": "2025-09-10",
    "dateTo": "2025-09-20",
    "search": "cleaning"
}

example_4 = {
  "city": "delhi",
  "skus": ["SKU1", "SKU2"],
  "page": 1,
  "pageSize": 2,
  "sortBy": "TOTAL STOCK",
  "sortOrder": "DESC",
  "dateFrom": "2025-09-01",
  "dateTo": "2025-09-30",
  "search": "baby wipes"
}


def test_api(example_data, example_name):
    """Test the API with example data"""
    print(f"\n=== {example_name} ===")
    print("Request JSON:")
    print(json.dumps(example_data, indent=2))
    
    try:
        response = requests.post(url, json=example_data)
        print(f"\nResponse Status: {response.status_code}")
        
        if response.status_code == 200:
            print("Response JSON:")
            print(json.dumps(response.json(), indent=2))
        else:
            print(f"Error: {response.text}")
    except requests.exceptions.ConnectionError:
        print("Error: Could not connect to the API. Make sure the server is running on http://localhost:8000")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    print("Stock Availability API Test Examples")
    print("=" * 50)
    
    # Test all examples
    test_api(example_1, "Example 1: Chennai toothbrush search")
    test_api(example_2, "Example 2: Baby products search")
    test_api(example_3, "Example 3: Cleaning products with date range")
    test_api(example_4, "Example 4: Delhi baby wipes search")

    print("\n" + "=" * 50)